File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {

    let temp = reserve.map((i) => i);
    
    reserve = reserve.filter((i) => !lost.includes(i));
    lost = lost.filter((i) => !temp.includes(i));
    
    reserve.sort();
    lost.sort();
    
    let answer = n - lost.length;
  
    for (let i = 0; i < reserve.length; i++) {
        for (let j = 0; j < lost.length; j++) {
            if (reserve[i]-1 == lost[j] || reserve[i]+1 == lost[j]){
                reserve[i] = -1;
                lost[j] = -1;
                answer += 1;
                break;
            }
        }
    }
 
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    
    res_, lost_ = [], []
    for i in lost:
        if i not in reserve: lost_.append(i)
    for i in reserve:
        if i not in lost: res_.append(i)
    
        
    answer = n - len(lost_)
    lost_.sort()
    res_.sort()
    
    myDict = dict()
    for i in res_: myDict[i] = 1
    
    for i in lost_:
        for j in (-1, 1):
            now = i+j
            if now in myDict and myDict[now]:
                answer += 1
                myDict[now] = 0
                break
    
    return answer